src/client/datascience/cellMatcher.ts

```diff
@@ -16,18 +17,30 @@ export class CellMatcher {
     private defaultCellMarker: string;
     private defaultCellMarkerExec: RegExp;
 
-    constructor(settings?: IJupyterSettings) {
+    constructor(settings?: IJupyterSettings, language?: string) {
+
+        if (language) {
```

Comment:
I think I'd rewrite this like so:

```suggestion
if (!language) {
   language = python
}
```

And don't use the active text editor here. It should be used in the caller. Otherwise this code has an implicit dependency on the window namespace that isn't necessary.

Reply:
How do you know that you want to use the active editor? There are cases where this is called on a non active editor

Reply:
So that's why I'd pass in the language only. Let the caller decided what the language is.

Reply:
Ah so you're saying, don't pass in a default value for language, just let it be an argument in the method and end it there?

Reply:
Yeah I think so.

Reply:
Make it the responsibility of the caller to determine it

Reply:
>  that wasn't done for any good reason, just don't know the best way to get the `TextDocument` object in each situation, so i defaulted to that.

Reply:
No worries. The calling code should be able to make a better decision I think about whether or not to use the active editor to determine the language. You might have to trace back a few levels though (in the callers) to determine what they're actually doing. I think some of the code in cellFactory.ts has some intermediary functions that you might need to pass the language through as well.

Reply:
So, sorry for the hand holding, but youre asking that I look through all instances of CellMatcher and see where a language argument is needed?

Reply:
Well all of them should need one, but each caller will get it in different ways. So yes, you'd have to look for all callers of CellMatcher (you should be able to use the same go to references for the constructor to find them).